sycl/source/detail/graph_impl.cpp

```diff
@@ -680,7 +681,10 @@ sycl::detail::pi::PiExtSyncPoint exec_graph_impl::enqueueNode(
 void exec_graph_impl::createCommandBuffers(
     sycl::device Device, std::shared_ptr<partition> &Partition) {
   sycl::detail::pi::PiExtCommandBuffer OutCommandBuffer;
-  sycl::detail::pi::PiExtCommandBufferDesc Desc{};
+  sycl::detail::pi::PiExtCommandBufferDesc Desc{
+      pi_ext_structure_type::PI_EXT_STRUCTURE_TYPE_COMMAND_BUFFER_DESC, nullptr,
+      pi_bool(Partition->MIsInOrderGraph & !MEnableProfiling),
```

Comment:
```suggestion
      pi_bool(Partition->MIsInOrderGraph && !MEnableProfiling),
```
Since we're dealing with `bool` here preference for `&&` since I don't believe `&` will perform short circuit evaluation.